third_party/2/werkzeug/wrappers.pyi

```diff
@@ -2,78 +2,85 @@
 #
 # NOTE: This dynamically typed stub was automatically generated by stubgen.
 
-from typing import Any
+from typing import (
+    Any, Iterable, Mapping, Optional, Sequence, Tuple, Type, Union,
+)
+
+from .datastructures import (
+    CombinedMultiDict, EnvironHeaders, Headers, ImmutableMultiDict,
+    MultiDict, TypeConversionDict,
+)
 
 class BaseRequest:
-    charset = ...  # type: Any
-    encoding_errors = ...  # type: Any
-    max_content_length = ...  # type: Any
-    max_form_memory_size = ...  # type: Any
-    parameter_storage_class = ...  # type: Any
-    list_storage_class = ...  # type: Any
-    dict_storage_class = ...  # type: Any
-    form_data_parser_class = ...  # type: Any
-    trusted_hosts = ...  # type: Any
+    charset = ...  # type: basestring
```

Comment:
@JelleZijlstra Okay, I will change attributes and return values to be typed as one of `str` or `unicode` depending on its typical uses.  In case of arguments, I will stick with `basestring`.